### PR TITLE
Run (pure) flake8 in CI; deprecate config for pytest-flake8

### PR DESCRIPTION
**Pull request by bobhy**
Adds `flake8` back into CI.  Runs `flake8` itself rather than the (dimly-recalled)  `pytest --flake8`, which has a different syntax for configuring lint exceptions.  Now a single list of lint exceptions can apply in developer's IDE and to CI.  (DRY)

### CODE REVIEW

**Comment by bobhy** [comment]
This PR is ready for review and merge.
I've verified that either pytest or flake8 can signal CI failure.

**Comment by bobhy** [comment]
This PR is focused on adding flake8 check back into CI, cleaning up the flake8 config in setup.cfg and removing the (now redundant) pytest-flake8 config.  It is a prerequisite for #3560 despite the PR number.

**Comment by bobhy** [comment]
Ready, once again, for review.  Please fill or kill before more bit rot sets in.

File: news/flake8_in_CI.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+* CI step to run flake8 after pytest.
+
+**Changed:**
+
+* Updated pytest_plugin for pytest 5.4 API, pip requirements for pytest>= 5.4
+
+**Deprecated:**
+
+* `pytest --flake8`_ now exits with error message to use flake8 instead.
```
Comment (laloch) [inline news/flake8_in_CI.rst:11]: I think, the trailing underscore after ``` `pytest --flake8` ``` shouldn't be here, right? Also, I think, you wanted to write ``` ``pytest --flake8`` ``` (with double backticks).

File: xonsh/environ.py [hunk 2]
```diff
@@ -366,9 +366,7 @@ def __str__(self):
         return str(self._d)
 
     def __repr__(self):
-        return "{0}.{1}(...)".format(
-            self.__class__.__module__, self.__class__.__name__, self._d
-        )
+        return "{0}.{1}(...)".format(self.__class__.__module__, self.__class__.__name__)
```
Comment (laloch) [inline xonsh/environ.py:369]: Why did you change the `__repr__` here? Does this belong to this patchset?

File: xonsh/environ.py [hunk 3]
```diff
@@ -1663,9 +1661,7 @@ def __str__(self):
...
```
Comment (laloch) [inline xonsh/environ.py:1664]: Same as above: Why did you change the `__repr__` here? Does this belong to this patchset?

File: news/flake8_in_CI.rst (hunk 1 above)
Comment (bobhy) [reply news/flake8_in_CI.rst:11]: I was going for a link highlight, which this does.  But I see that's not how it's done in other news files....

File: xonsh/environ.py (hunk 2 above)
Comment (bobhy) [reply xonsh/environ.py:369]: I was getting flake8 F523 "'...'.format(...) has unused arguments" errors.  Same for change below.

File: xonsh/environ.py (hunk 3 above)
Comment (bobhy) [reply xonsh/environ.py:1664]: *idem*

File: xonsh/environ.py (hunk 2 above)
Comment (laloch) [reply xonsh/environ.py:369]: Yes, makes sense.

File: news/flake8_in_CI.rst (hunk 1 above)
Comment (laloch) [reply news/flake8_in_CI.rst]: Thanks!

**Comment by laloch** [comment]
Thanks @bobhy!
